docs/how-to/mgc-setup.md

```diff
@@ -0,0 +1,145 @@
+
+To manage the creation of DNS records, MGC uses [ManagedZone](https://docs.kuadrant.io/multicluster-gateway-controller/docs/how-to/managedZone/) resources. A `ManagedZone` can be configured to use DNS Zones on either AWS (Route53), and GCP. We will now create a ManagedZone on the cluster using AWS credentials.
+
+First, export the [environment variables detailed here](https://docs.kuadrant.io/multicluster-gateway-controller/docs/getting-started/#config) in a terminal session.
+
+Next, create a secret and configMap containing the AWS credentials and config [TODO: is the ConfigMap needed here?]:
+
+```bash
+cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: Secret
+metadata:
+  name: mgc-aws-credentials
+  namespace: multicluster-gateway-controller-system
+type: "kuadrant.io/aws"
+stringData:
+  AWS_ACCESS_KEY_ID: ${MGC_AWS_ACCESS_KEY_ID}
+  AWS_SECRET_ACCESS_KEY: ${MGC_AWS_SECRET_ACCESS_KEY}
+  AWS_REGION: ${MGC_AWS_REGION}
+EOF
+
+cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: mgc-controller-config
+  namespace: multicluster-gateway-controller-system
+data:
+  AWS_DNS_PUBLIC_ZONE_ID: ${MGC_AWS_DNS_PUBLIC_ZONE_ID}
+  ZONE_ROOT_DOMAIN: ${MGC_ZONE_ROOT_DOMAIN}
+  LOG_LEVEL: "1"
+EOF
+```
+
+A `ManagedZone` can then be created:
+
+```bash
+cat <<EOF | kubectl apply -f -
+apiVersion: kuadrant.io/v1alpha1
+kind: ManagedZone
+metadata:
+  name: mgc-dev-mz
+  namespace: multi-cluster-gateways
+spec:
+  id: ${MGC_AWS_DNS_PUBLIC_ZONE_ID}
+  domainName: ${MGC_ZONE_ROOT_DOMAIN}
+  description: "Dev Managed Zone"
+  dnsProviderSecretRef:
+    name: mgc-aws-credentials
+    namespace: multicluster-gateway-controller-system
+    type: AWS
+EOF
+```
+
+You can now verify that the `ManagedZone` has been created and is in a ready state:
+
+```bash
+kubectl get managedzone -n multi-cluster-gateways
+```
+```
+NAME         DOMAIN NAME      ID                                  RECORD COUNT   NAMESERVERS                                                                                         READY
+mgc-dev-mz   ef.hcpapps.net   /hostedzone/Z06419551EM30QQYMZN7F   2              ["ns-1547.awsdns-01.co.uk","ns-533.awsdns-02.net","ns-200.awsdns-25.com","ns-1369.awsdns-43.org"]   True
+```
+
+## Creating a Cert Issuer
+
+We will now create a `ClusterIssuer` to be used with `cert-manager`. For simplicity, we will create a self-signed cert issuer here, but [other issuers can also be configured](https://cert-manager.io/docs/configuration/).
+
+```bash
+cat <<EOF | kubectl apply -f -
+apiVersion: cert-manager.io/v1
+kind: ClusterIssuer
+metadata:
+  name: glbc-ca
+  namespace: cert-manager
+spec:
+  selfSigned: {}
+EOF
+```
+
+Verify that the `clusterIssuer` is ready:
+
+```bash
+kubectl get clusterissuers -n cert-manager
+```
+```
+NAME      READY   AGE
+glbc-ca   True    3s
+```
+
+## Next Steps
+
+Now that you have MGC installed and configured in your hub cluster, you can now continue with any of these follow-on guides:
```

Comment:
I think we are missing the setup of the gatewayclass

Reply:
I added a section on creating the `GatewayClass`, but just thinking now would it be better to remove this section and apply the `GatewayClass` in the kustomize diectory that deploys MGC, add-on manager,  etc?

Reply:
I ran into an issue doing this because the gateway API CRDs and GatewayClass are applied together and because of this the CRDs aren't ready in time for the GatewayClass creation. Latest commit still separates creation of GatewayClass, but the section is a bit shorter and the resource is remotely applied.

Another option we could possibly take with this would be to:
1. Install gateway API CRDs first
2. Then apply the MGC, add-on manager, GatewayClass, etc. using kustomize

Reply:
I think installing Gateway API at a particular version is a pre-req so installing it first makes sense +1